ha/setup/ha_setup.py

```diff
@@ -0,0 +1,315 @@
+        args = parser.parse_args(argv)
+        return args.command(args)
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """
+        Add Command args for parsing.
+        """
+        parser1 = parser.add_parser(cls.name, help='setup %s' % name)
+        parser1.add_argument('--config', help='Config URL')
+        parser1.add_argument('--plan', nargs="?", default=[], help='Test plan')
+        parser1.add_argument('args', nargs='*', default=[], help='args')
+        parser1.set_defaults(command=cls)
+
+class PostInstallCmd(Cmd):
+    """
+    PostInstall Setup Cmd
+    """
+    name = "post_install"
+
+    def __init__(self, args: dict):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process post_install command.
+        """
+
+        # Create a directory and copy config file
+        if os.path.exists(const.HA_CONFIG_FILE):
+            os.remove(const.HA_CONFIG_FILE)
+        os.makedirs(const.CONFIG_DIR, exist_ok=True)
+        shutil.copyfile(const.SOURCE_CONFIG_FILE, const.HA_CONFIG_FILE)
+        Log.info("Copied HA config file.")
+        # Pre-requisite checks are done here.
+        # Make sure the pacemaker, corosync and pcs packages have been installed
+        PkgV().validate('rpms', const.PCS_CLUSTER_PACKAGES)
+
+class ConfigCmd(Cmd):
+    """
+    Setup Config Cmd
+    """
+    name = "config"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process config command.
+        """
+        # Read machine-id and using machine-id read minion name from confstore
+        # This minion name will be used for adding the node to the cluster.
+        nodelist = []
+        command = "cat /etc/machine-id"
+        machine_id, err, rc = self._execute.run_cmd(command, check_error=True)
+        Log.info(f"Read machine-id. Output: {machine_id}, Err: {err}, RC: {rc}")
+        minion_name = Conf.get(self._index, f"cluster.server_nodes.{machine_id.strip()}")
+        nodelist.append(minion_name)
+        # The config step will be called from primary node alwasys,
+        # see how to get and use the node name then.
+
+        # Read cluster name and cluster user
+        cluster_name = Conf.get(self._index, 'corosync-pacemaker.cluster_name')
+        cluster_user = Conf.get(self._index, 'corosync-pacemaker.user')
+
+        # Read cluster user password and decrypt the same
+        cluster_id = Conf.get(self._index, 'cluster.cluster_id')
+        cluster_secret = Conf.get(self._index, 'corosync-pacemaker.secret')
+        key = Cipher.generate_key(cluster_id, cluster_user)
+        cluster_secret = Cipher.decrypt(key, cluster_secret.encode('ascii')).decode()
+
+        Log.info(f"Creating cluster: {cluster_name} with node: {minion_name}")
+        # TBD: Call script for creating and configuring the cluster
```

Comment:
This piece of code is still missing?

Reply:
This will come after other PR (#283) where the rules are defined gets merged. I can add if the other PR is merged earlier than this one. Or if this PR gets merged first then we can update PR #283.

Reply:
done